controllers/machine_controller_phases.go

```diff
@@ -78,6 +78,13 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, m *clusterv1.
 	obj, err := external.Get(r.Client, ref, m.Namespace)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
+			if m.Status.InfrastructureReady && m.Status.GetTypedPhase() == clusterv1.MachinePhaseRunning {
```

Comment:
We shouldn't have logic comparing against the Machine "Phase", it is meant to only be a user friendly display value.

This check should also probably be pushed up to where reconcileExternal is called for the InfrastructureRef, since a missing Bootstrap.ConfigRef should not result in an error.

Reply:
I added the machine phase to make sure this block of code is executed only once.